cmd/deploy_test.go

```diff
@@ -1372,18 +1368,15 @@ func testRegistry(cmdFn commandConstructor, t *testing.T) {
 			expectedImage:    "registry.example.com/charlie/f:latest",
 		},
 		{
-			// Image flag takes highest precidence, but is an override such that the
-			// resultant image member is updated but the registry member is not
-			// updated (subsequent builds will not be affected).
+			// Image flag sets the image directly when no registry is present
 			name: "image flag overrides",
```

Comment:
I think in this test we want to test exactly the case where image DOES override registry since registry is not being provided via flag but rather its already been given previously.

This test seems to to to cover the case when you 
1. `func deploy --registry ...`
... now `f.registry` exists in `func.yaml` and is not empty 
2. `func deploy --image <different image>`

we want this case to work -> override the image from `registry` thats been used from `1.` with `image` from `2.` 
the `func.yaml` registry is NOT overriden and kept as is from `1.` but the new image is used in deployment because `--image` was provided with different value.